lib/screens/admin_order_detail_page.dart

```diff
@@ -0,0 +1,585 @@
+            Text(
+              widget.order.getOrderStatusText(),
+              style: TextStyle(
+                color: kPaletteWhite,
+                fontSize: 16.0,
+                fontWeight: FontWeight.w600,
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget _buildOrderItemList() {
+    return Container(
+      padding: EdgeInsets.symmetric(horizontal: 16.0),
+      child: StreamBuilder<QuerySnapshot>(
+        stream: Firestore.instance
+            .collection(constants.DBCollections.products)
+            .snapshots(),
+        builder: (context, snapshot) {
+          if (!snapshot.hasData) {
+            return LinearProgressIndicator();
+          }
+
+          return Column(
+            children: widget.order.items.map( (orderItem) {
+              DocumentSnapshot productSnapshot = snapshot.data.documents.firstWhere(
+                (snapshot) => snapshot.documentID == orderItem.productId
+              );
+              return _buildOrderItemListRow(
+                orderItem, Product.fromSnapshot(productSnapshot)
+              );
+            }).toList(),
+          );
+        }
+      ),
+    ); 
+  }
+
+  Widget _buildOrderItemListRow(OrderItem orderItem, Product product) {
+    return Container(
+      decoration: BoxDecoration(
+        border: Border(
+          bottom: BorderSide(color: const Color(0xFFA2A2A2)),
+        ),
+      ),
+      padding: EdgeInsets.symmetric(vertical: 16.0),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        children: <Widget>[
+          _buildProductImage(product),
+          _buildOrderItemTextInfo(orderItem, product),
+          _buildOrderItemQuantity(orderItem, product),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildProductImage(Product product) {
+    Widget productImage;
+    if (product.imageUrls?.isEmpty ?? true) {
+      productImage = Image.memory(
+        kTransparentImage,
+        height: 85,
+        width: 85,
+      );
+    } else {
+      productImage = FadeInImage.memoryNetwork(
+        fit: BoxFit.cover,
+        height: 85,
+        width: 85,
+        placeholder: kTransparentImage,
+        image: product.imageUrls.first,
+      );
+    }
+
+    return ClipRRect(
+      borderRadius: BorderRadius.circular(12.0),
+      child: productImage,
+    );
+  }
+
+  Widget _buildOrderItemTextInfo(OrderItem orderItem, Product product) {
+    return Container(
+      height: 75,
+      padding: EdgeInsets.only( right: 50.0 ),
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: <Widget>[
+          _buildProductName(product),
+          _buildOrderItemColorSize(orderItem, product),
+          _buildOrderItemPrice(orderItem, product),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildProductName(Product product) {
+    return Container(
+      child: Text(
```

Comment:
possible overflow issue when product name is too long. please use some [overflow](https://stackoverflow.com/a/44580054) measure like:

`overflow: TextOverflow.ellipsis,`